Sources/SwiftGRPCNIO/CallHandlers/ClientStreamingCallHandler.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import SwiftProtobuf
+import NIO
+import NIOHTTP1
+
+/// Handles client-streaming calls. Forwards incoming messages and end-of-stream events to the observer block.
+///
+/// - The observer block is implemented by the framework user and fulfills `context.responsePromise` when done.
+public class ClientStreamingCallHandler<RequestMessage: Message, ResponseMessage: Message>: BaseCallHandler<RequestMessage, ResponseMessage> {
+  public typealias EventObserver = (StreamEvent<RequestMessage>) -> Void
+  private var eventObserver: EventLoopFuture<EventObserver>?
+  
+  private var context: UnaryResponseCallContext<ResponseMessage>?
```

Comment:
Same in these other files

Reply:
No; the context should live for as long as the handler does. The closure, however, has a much shorter lifetime.